daisy/workflow/step_create_instances.go

```diff
@@ -143,7 +184,7 @@ func (c *CreateInstances) run(w *Workflow) error {
 			for i, sourceDisk := range ci.AttachedDisks {
 				var disk *resource
 				var err error
-				if isLink(sourceDisk) {
+				if isLink(sourceDisk) && diskValid(w, sourceDisk) {
 					// Real link.
 					inst.AddPD("", sourceDisk, false, i == 0)
 				} else if disk, err = w.getDisk(sourceDisk); err == nil {
```

Comment:
This needs to be done at validation time. To do that we need to store project and zone in nameSet, re reuse refmap instead but with a noop call so the compute service isn't called in validate.